### PR TITLE
[config] Label Clang 18.0-19.trunk SA & Tidy checkers

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by sylvestre** [comment]
Failure is this one but not sure why it is my fault:

```
Run set +e # Do not hard exit on an erroring call!
Checkers REMOVED from upstream, with CodeChecker still assigning labels:
Skipped! Specify '--report-removed' to execute.

Checkers ADDED to upstream, without labels in CodeChecker:
 + bugprone-chained-comparison
 + bugprone-crtp-constructor-accessibility
 + bugprone-unused-local-non-trivial-variable
 + cppcheck-constParameterPointer
 + cppcheck-constParameterReference
 + cppcheck-constVariablePointer
 + cppcheck-constVariableReference
 + cppcheck-eraseIteratorOutOfBounds
 + cppcheck-eraseIteratorOutOfBoundsCond
 + cppcheck-knownPointerToBool
 + cppcheck-pointerOutOfBoundsCond
 + hicpp-ignored-remove-result
 + modernize-use-designated-initializers
 + optin.core.EnumCastOutOfRange
 + readability-use-std-min-max
Coverage check returned: 8.
Error: Process completed with exit code 8.
```

**Comment by whisperity** [comment]
> Failure is this one but not sure why it is my fault:

@sylvestre You changed a label configuration file, so the CI job that checks whether the label configuration file covers the nightly build of the respective analysers were executed. 

For example, it caught that you made a typo in the config file. 🙂

File: config/labels/analyzers/clang-tidy.json [hunk 1]
```diff
@@ -696,6 +696,13 @@
       "sei-cert:mem51-cpp",
       "severity:MEDIUM"
     ],
+    " bugprone-unused-local-non-trivial-variable": [
+      "doc_url:https://clang.llvm.org/extra/clang-tidy/checks/bugprone/unused-local-non-trivial-variable.html",
+      "profile:default",
```
[the remark this answers is not in the document]
Comment (sylvestre) [reply config/labels/analyzers/clang-tidy.json:701]: copy and paste from bugprone-unused-raii
let me remove that

[the remark this answers is not in the document]
Comment (whisperity) [reply config/labels/analyzers/clang-tidy.json]: If the check is of good quality I've no argument against including it into default!

Maybe it'd be worthwhile to run some tests of these checks on a large enough project. I can throw them into our CI hopefully tomorrow. ☺

**Comment by whisperity** [comment]
@nagydonat @discookie Can you please describe the `cplusplus.ArrayDelete` checker for us? It went out of alpha following llvm/llvm-project#83985, but we did NOT have a label/guideline/documentation/severity/etc. configuration for the check's alpha version, and now should add one for the non-alpha one. What are your suggestions?

**Comment by whisperity** [comment]
@NagyDonat @gamesh411 Same comment for `optin.core.EnumCastOutOfRange`, following llvm/llvm-project#67157 & llvm/llvm-project#68191. For the previous `alpha.cplusplus.EnumCastOutOfRange`, we had

```json
    "alpha.cplusplus.EnumCastOutOfRange": [
      "doc_url:https://clang.llvm.org/docs/analyzer/checkers.html#alpha-cplusplus-enumcastoutofrange-c",
      "profile:extreme",
      "profile:sensitive",
      "severity:MEDIUM"
    ],
```

**Comment by NagyDonat** [comment]
`cplusplus.ArrayDelete` is documented at `https://clang.llvm.org/docs/analyzer/checkers.html#cplusplus-arraydelete-c` and it is specific to C++. I'm not familiar with the meaning of the potential `severity` and `profile` values in codechecker, so I cannot help you with that. Maybe @gamesh411 could categorize these checkers in that respect?

**Comment by whisperity** [comment]
> I'm not familiar with the meaning of the potential `severity` and `profile` values in codechecker, so I cannot help you with that.

@NagyDonat These are the official descriptions for these values. I'm not inherently sure how strictly we adhere to them.

https://github.com/Ericsson/codechecker/blob/e2d7df4f7b8d5e8a7f315e24fc94aeb756f71771/config/labels/descriptions.json#L2-L16

Basically the crucial issue about adding something to `profile:default` is that `CodeChecker analyze` will run them by default (hence the name), which is bad if the checker is throwing copious amounts of false positives or crashing left, right, and centre. The worse the checker is (in terms of user experience and correctness), the more it is pushed out on the `default -> sensitive -> extreme` spectrum.

Is it good to have this checker enabled by default for users? It's not part of `core.` but it sounds reasonable. Are the results sensible?

What happens if we default-enable a `cplusplus.Foo` CSA checker on a **C** source file?

**Comment by whisperity** [comment]
@gamesh411 Thanks! I'll update the patch soon and then I think we will be done with adding the new classifications!

**Review by bruntib — APPROVED** [review]

**Pull request merged**